wordpress-mobile/WordPress-iOS
# [JP Plugin] Enable feature flag and add release notes

Refs: pcdRpT-1Im-p2, qk5s4q4mZsXknwhMWo2vIY-fi-8221%3A20788
Resolves: #20020 
Depends on: #20220 👈🏼 Please review this first.

<img width="1222" alt="Screenshot 2023-02-28 at 23 43 25" src="https://user-images.githubusercontent.com/1299411/221919907-c2ea751b-0691-452c-9f8c-4c4e1fa710fa.png">

This PR enables the support for individual Jetpack plugins **in the Jetpack app**.

### Feature Summary

1. In the Jetpack app, when they switch to a self-hosted site that's connected only through individual Jetpack plugins, we'll show a pop-up that offers the full plugin. We'll only show the pop-up once per site.
2. Through this pop-up, users can go through the flow to install the full plugin for their site. After they've successfully installed the plugin, the app's core functionalities are expected to work normally.
3. If the user didn't install, we'll show a card in the dashboard and the site menu. Users can tap on the card to see the pop-up again—and they will be able to go through the plugin installation flow from there.
5. Users can also dismiss this card by tapping on the card's contextual menu. Hiding either card will remove the card from both tabs. This is also an irreversible process; there's no way to make the card show again after it's hidden. Unless the user reinstalls the Jetpack app.

For more details, please refer to pcdRpT-1Im-p2 and pctCYC-An-p2.

## To Test

- Prepare a self-hosted site, and install an individual Jetpack plugin (e.g. Jetpack Search).
- If you have the full Jetpack plugin installed or activated for the site, please deactivate and uninstall that plugin.
- Create a Jetpack Connection to your WordPress.com account.
- Open the Jetpack app, and log in to your WordPress.com account.
- Switch to the self-hosted site.
- 🔎 Verify that the pop-up should be shown right after switching.
- 🔎 Verify that when the pop-up is dismissed, you can see the Jetpack plugin installation cards in the dashboard & site menu.
- Tap the "Learn More" button on one of the cards.
- Now, tap "Install the full plugin", and then "Continue".
- When the installation completes, tap "Done".
- 🔎 Verify that the cards are no longer shown.
- With a web browser, go to the Plugins page on your self-hosted site (`/wp-admin/plugins.php`)
- 🔎 Verify that the Jetpack plugin is installed and activated.

## Regression Notes
1. Potential unintended areas of impact
  - This feature should only be enabled ONLY in the Jetpack app, and for sites with ONLY individual Jetpack plugins. If you see this feature in the WordPress app, or enabled for other sites that don't fall into the criteria, it's likely an unintended side effect.
  - Some changes were made to the Jetpack Remote Install flow so it can be reused for this feature.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes, and added unit tests where it makes sense.

7. What automated tests I added (or what prevented me from doing so)
Added unit tests (in the previous related PRs).

PR submission checklist:
